core/src/elo/elo.service.ts

```diff
@@ -261,4 +262,54 @@ export class EloService {
         const seriesTypeStr = series.matchParent.fixture ? "fixture" : series.matchParent.scrimMeta ? "scrim" : "unknown";
         return `\`seriesId=${seriesId}\` ${seriesTypeStr ? `(${seriesTypeStr})` : ""} successfully marked \`fullNcp=${isNcp}\` with updated elo, and all connected replays had their elo updated.${numReplays && dummiesNeeded ? ` **${dummiesNeeded} dummy replay(s)** were added to the series.` : ""}`;
     }
+
+    async prepMigrationData(): Promise<NewPlayer[]> {
+        // Run queries, first refreshing the materialized view
+        this.logger.verbose("Refreshing materialized view.");
+        // await this.dataSource.manager.query("REFRESH MATERIALIZED VIEW mledb.v_current_elo_values");
+
+        // Then querying from it
+        /* eslint-disable @typescript-eslint/no-unsafe-assignment,
+        @typescript-eslint/no-explicit-any */
+        this.logger.verbose("Querying the materialized view.");
+        const rawData: any[] = await this.dataSource.manager.query("SELECT player_id, elo, league, salary, name FROM mledb.v_current_elo_values");
+
+        this.logger.verbose(`Elo Service, querying migration data: ${JSON.stringify(rawData[0])}`);
+        // Now, we build a NewPlayer instance for each row of data returned from
+        // the query
+        const output: NewPlayer[] = rawData.map(r => ({
+            id: r.player_id,
+            name: r.name,
+            salary: r.salary,
+            skillGroup: this.skillGroupStringToInt(r.league as string), // Map strings to numbers? not sure yet
+            elo: r.elo,
+        }));
+
+        this.logger.verbose(`Object build from query: ${JSON.stringify(output[0])}`);
+
+        return output;
+    }
+
+    skillGroupStringToInt(skillGroup: string): number {
+        switch (skillGroup) {
+            case "FOUNDATION": {
+                return 1;
+            }
+            case "ACADEMY": {
+                return 2;
+            }
+            case "CHAMPION": {
+                return 3;
+            }
+            case "MASTER": {
+                return 4;
+            }
+            case "PREMIER": {
+                return 5;
+            }
+            default: {
+                return 6;
```

Comment:
Throw new error here because if it doesn't exist we don't want random things happening

Reply:
Leaving this open for now, as there do exist players with skillGroup "UNKNOWN" in the mledb schema (and thus in the view that we're pulling this data from). TODO is in the code to review this once the MLEDB->Sprocket migration is complete, and the view is updated to pull from players in the sprocket schema.